File: WinTexRFID/Backup/WaitForm.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using System.Threading;

namespace UHFAPP
{
    public partial class WaitForm : Form
    {

        private static WaitForm loadingForm = null;
        public WaitForm() {
            InitializeComponent();
          
        }
     
        /// <summary>
        /// 显示加载
        /// </summary>
        public static void ShowLoading()
        {
            try
            {
                Thread loadingTh = new Thread(new ThreadStart(delegate
                {
                    loadingForm = new WaitForm();
                    loadingForm.StartPosition = FormStartPosition.CenterParent;
                    Application.Run(loadingForm);
                    loadingForm.Dispose();
                    loadingForm = null;
                    loadingTh = null;
                }));
                loadingTh.IsBackground = true;
                loadingTh.Priority = ThreadPriority.Highest;
                loadingTh.Start();                 
            }
            catch
            { }
        }
        /// <summary>
        /// 隐藏加载
        /// </summary>
        public static void HideLoading()
        {
            try
            {
                if (loadingForm != null)
                {
                    loadingForm.Invoke(new EventHandler(delegate
                    {
                        loadingForm.Close();
                    }));
                }
            }
            catch
            { }
        }
    }


}

File: WinTexRFID/WaitForm.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports System.Threading

Namespace UHFAPP
	Partial Public Class WaitForm
		Inherits Form

		Private Shared loadingForm As WaitForm = Nothing
		Public Sub New()
			InitializeComponent()

		End Sub

		''' <summary>
		''' 显示加载
		''' </summary>
		Public Shared Sub ShowLoading()
			Try
				Dim loadingTh As New Thread(New ThreadStart(Sub()
					loadingForm = New WaitForm()
					loadingForm.StartPosition = FormStartPosition.CenterParent
					Application.Run(loadingForm)
					loadingForm.Dispose()
					loadingForm = Nothing
					loadingTh = Nothing
				End Sub))
				loadingTh.IsBackground = True
				loadingTh.Priority = ThreadPriority.Highest
				loadingTh.Start()
			Catch
			End Try
		End Sub
		''' <summary>
		''' 隐藏加载
		''' </summary>
		Public Shared Sub HideLoading()
			Try
				If loadingForm IsNot Nothing Then
					loadingForm.Invoke(New EventHandler(Sub()
						loadingForm.Close()
					End Sub))
				End If
			Catch
			End Try
		End Sub
	End Class


End Namespace
